akka-remote/src/main/scala/akka/remote/artery/SendQueue.scala

```diff
@@ -119,8 +119,10 @@ private[remote] final class SendQueue[T] extends GraphStageWithMaterializedValue
         val q = producerQueue
         if (q eq null) throw new IllegalStateException("offer not allowed before injecting the queue")
         val result = q.offer(message)
-        if (result && needWakeup)
+        if (result && needWakeup) {
+          needWakeup = false
```

Comment:
I was worried about races here (since the only writer of this field was the queue consumer side so far), but it seems like it is impossible. The only way for a missing wakeup is that the following conditions hold at the same time:
 - there is a message in the queue
 - the consumer is not pulling
 - there is no enqueued wakeup signal

As far as I see that cannot happen because
 - there is a message in the queue implies that `offer` has been called, which implies that `needWakeup` was checked soon after
 - the consumer is not pulling implies `needWakeup` was set to true because it sets this variable *before* it stops pulling (it pulls one more after the set)
 - there is no enqueued wakeup signal implies that `needWakeup` was seen as false after an offer, but that implies that either the consumer did not stop pulling, or another offer has enqueued the wakeup already (or will enqueue soon, if we were concurrently executing)